client/src/main/java/org/spine3/client/QueryOperator.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.client;
+
+import com.google.common.base.Objects;
+import com.google.protobuf.Timestamp;
+
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.spine3.time.Timestamps2.isLaterThan;
+
+/**
+ * An enumeration of all supported value comparison operators applicable to the Entity Columns.
+ *
+ * <p>Usage of this enumeration when building a {@link Query} is unnecessary. It's effectively
+ * {@linkplain org.spine3.annotation.Internal internal} except for the
+ * {@linkplain org.spine3.annotation.SPI service provider interface implementors}.
+ *
+ * <p>Even though the type provides an interface for performing the in-memory comparisons by the
+ * listed operators, the main purpose is to show the strategy of the comparison, not implement it.
+ */
+public enum QueryOperator {
+
+    /**
+     * Equality operator ({@code =}).
+     */
+    EQUAL("=") {
+        /**
+         * {@inheritDoc}
+         *
+         * <p>Equality operator supports {@code null} values and also returns {@code true} if both
+         * the operands are equal to {@code null}.
+         */
+        @Override
+        <T> boolean compare(@Nullable T left, @Nullable T right) {
+            return Objects.equal(left, right);
+        }
+    },
+
+    /**
+     * Comparison operator stating that the stored value is greater then ({@code >})
+     * the passed value.
+     */
+    GREATER_THEN(">") {
```

Comment:
Can we make it `GREATER_OR_EQUAL` and `LESS_OR_EQUAL`?
Also, let's discuss if we should do that in the scope of this PR.